jetty-core/jetty-util/src/main/java/org/eclipse/jetty/util/resource/Resource.java

```diff
@@ -151,7 +126,10 @@ else if (urlString.startsWith("jar:"))
      */
     public static Resource newResource(String resource) throws IOException
     {
-        return newResource(resource, __defaultUseCaches);
+        URI uri = URI.create(resource);
+        FileSystem fileSystem = FileSystems.newFileSystem(uri, new HashMap<>());
+        Path path = Paths.get(uri);
+        return newResource(path);
```

Comment:
Does this work with non path URIs (like `http:` or `jrt:`)?   Also what about non URIs?  Currently a use can pass in something like "C:\my\webapp.war" and since that fails as a URL we try it as a PathResource.   See the `newResource(String resource, boolean useCaches)` impl for details.

Reply:
It works with registered schemes and by default the JVM registers `file:`, `jar:` and `jrt:`. So no, a `http:` URI won't work but you're right that the logic used for `URLResource` that tries to fallback to a `PathResource` should be added to support direct path names.